### PR TITLE
Variant for enabling tests, versions, cleanup

### PR DESCRIPTION
**Pull request by raovgarimella**
Added variant for enabling tests
Added option to retrieve variants
Split 'parallel=some_partitioner' variant into 'parallel=true/false' and 'partitioner=some_partitioner'
Added conflicts
General cleanup

### CODE REVIEW

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 1]
```diff
@@ -24,33 +24,57 @@ class Mstk(CMakePackage):
     homepage = "https://github.com/MeshToolkit/MSTK"
     git      = "https://github.com/MeshToolkit/MSTK"
 
-    maintainers = ['julienloiseau']
+    maintainers = ['raovgarimella','julienloiseau']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]:
```suggestion
    maintainers = ['raovgarimella', 'julienloiseau']
```
flake8

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 2]
```diff
@@ -24,33 +24,57 @@ class Mstk(CMakePackage):
...
 
+    def url_for_version(self, version):
+        url = 'https://github.com/MeshToolkit/MSTK/archive/{0}.tar.gz'
+        return url.format(version)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]: I would drop this function and remove the `v` from the version numbers below, that should work.

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 3]
```diff
@@ -24,33 +24,57 @@ class Mstk(CMakePackage):
...
+    version('v3_0_2', sha256='1cdae8cad0cb65991854bb7667f5a08e6c9ea23f41e2dca7f3484c4c0b8f9a90')
+    version('v3_0_1', sha256='ac63063c861e605ef9060f726568600f9ad367d6b7e2a5664bc97e766d188f4d')
+    version('v3_0_0', sha256='7f1b3ad68b397d5487de87122c642b48fb76f656ac68a733bf8a0195fddbaeac')
 
-    variant('parallel', default='none', description='Enable Parallel Support',
-            values=('none', 'metis', 'zoltan', 'parmetis'), multi=True)
     variant('exodusii', default=False, description='Enable ExodusII')
-    variant('use_markers', default=True, description="Enable MSTK to use markers")
+    variant('use_markers', default=True, description='Enable use of markers')
+    variant('enable_tests', default=False, description='Enable unit testing')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]: Drop this variant, there is a better way to enable unit tests

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 4]
```diff
@@ -24,33 +24,57 @@ class Mstk(CMakePackage):
...
-    depends_on("metis", when='parallel=parmetis')
 
     # Exodusii variant
     # The default exodusii build with mpi support
-    # It includes netcdf which includes hdf5
-    depends_on("exodusii", when='+exodusii')
+    depends_on('exodusii', when='+exodusii')
+
+    # Unit testing variant
+    depends_on('unittest-cpp', when='+enable_tests')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]:
```suggestion
    depends_on('unittest-cpp', type='test')
```

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 5]
```diff
@@ -60,34 +84,35 @@ def cmake_args(self):
...
+
 
         # ExodusII variant
         if '+exodusii' in self.spec:
             options.append('-DENABLE_ExodusII=ON')
         else:
             options.append('-DENABLE_ExodusII=OFF')
 
+        # Unit test variant
+        if '+enable_tests' in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]:
```suggestion
        if self.run_tests:
```

File: var/spack/repos/builtin/packages/mstk/package.py (hunk 1 above)
Comment (raovgarimella) [reply var/spack/repos/builtin/packages/mstk/package.py]: fixed

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 6]
```diff
@@ -60,34 +78,34 @@ def cmake_args(self):
...
-            options.append("-DENABLE_METIS=ON")
-            options.append("-DENABLE_ZOLTAN=ON")
-            options.append('-DZOLTAN_NEEDS_ParMETIS=ON')
+        if 'partitioner=none' in self.spec:
+            options.append('-DENABLE_METIS=OFF')
+            options.append('-DENABLE_ZOLTAN=OFF')
         else:
-            if "zoltan" in self.spec:
-                options.append("-DENABLE_ZOLTAN=ON")
+            if 'zoltan' or 'all' in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]:
```suggestion
            if 'zoltan' in self.spec:
```

File: var/spack/repos/builtin/packages/mstk/package.py [hunk 7]
```diff
@@ -60,34 +78,34 @@ def cmake_args(self):
...
+                options.append('-DENABLE_ZOLTAN=ON')
             else:
-                options.append("-DENABLE_ZOLTAN=OFF")
-            if "metis" in self.spec:
-                options.append("-DENABLE_METIS=ON")
+                options.append('-DENABLE_ZOLTAN=OFF')
+            if 'metis' or 'all' in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mstk/package.py]:
```suggestion
            if 'metis' in self.spec:
```

**Pull request merged**